source/priority-queue.ts

```diff
@@ -34,6 +34,10 @@ export default class PriorityQueue implements Queue<PriorityQueueOptions> {
 		return item && item.run;
 	}
 
+	filter(options: Partial<PriorityQueueOptions>): Array<{}> {
```

Comment:
The return value should be the same as the queue elements. So we need to make this class generic and return `Element[]`. And use `Element` in https://github.com/sindresorhus/p-queue/pull/94/files#diff-b581b23520a87f814e2b68a5e5432ee2R10 too.